src/cursor/find_cursor.ts

```diff
@@ -0,0 +1,390 @@
+      //       the value should only be saved in those cases.
+      if (response.cursor) {
+        this[kNumReturned] = response.cursor.firstBatch.length;
+      } else {
+        this[kNumReturned] = response.documents ? response.documents.length : 0;
+      }
+
+      // TODO: NODE-2882
+      callback(undefined, { server: findOperation.server, session, response });
+    });
+  }
+
+  /** @internal */
+  _getMore(batchSize: number, callback: Callback<Document>): void {
+    // NOTE: this is to support client provided limits in pre-command servers
+    const numReturned = this[kNumReturned];
+    if (numReturned) {
+      const limit = this[kBuiltOptions].limit;
+      batchSize =
+        limit && limit > 0 && numReturned + batchSize > limit ? limit - numReturned : batchSize;
+
+      if (batchSize <= 0) {
+        return this.close(callback);
+      }
+    }
+
+    super._getMore(batchSize, (err, response) => {
+      if (err) return callback(err);
+
+      // FIXME: wrap this in some logic to prevent it from happening if we don't need this support
+      if (response) {
+        this[kNumReturned] = this[kNumReturned] + response.cursor.nextBatch.length;
+      }
+
+      callback(undefined, response);
+    });
+  }
+
+  /**
+   * Get the count of documents for this cursor
+   *
+   * @param applySkipLimit - Should the count command apply limit and skip settings on the cursor or in the passed in options.
+   */
+
+  count(): Promise<number>;
+  count(callback: Callback<number>): void;
+  count(options: CountOptions): Promise<number>;
+  count(options: CountOptions, callback: Callback<number>): void;
+  count(
+    options?: CountOptions | Callback<number>,
+    callback?: Callback<number>
+  ): Promise<number> | void {
+    if (typeof options === 'boolean') {
+      throw new TypeError('Invalid first parameter to count');
+    }
+
+    if (typeof options === 'function') (callback = options), (options = {});
+    options = options || {};
+
+    return executeOperation(
+      this.topology,
+      new CountOperation(this.namespace, this[kFilter], {
+        ...this[kBuiltOptions], // NOTE: order matters here, we may need to refine this
+        ...this.cursorOptions,
+        ...options
+      }),
+      callback
+    );
+  }
+
+  /**
+   * Execute the explain for the cursor
+   *
+   * @param callback - The result callback.
+   */
+  explain(): Promise<Document>;
+  explain(callback: Callback<Document>): void;
+  explain(callback?: Callback<Document>): Promise<Document> | void {
+    return executeOperation(
+      this.topology,
+      new FindOperation(undefined, this.namespace, this[kFilter], {
+        ...this[kBuiltOptions], // NOTE: order matters here, we may need to refine this
+        ...this.cursorOptions,
+        explain: true
+      }),
+      callback
+    );
+  }
+
+  /** Set the cursor query */
+  filter(filter: Document): this {
+    this[kFilter] = filter;
+    return this;
+  }
```

Comment:
Should this check if the cursor has started and throw?

Reply:
yes, I'll add that to the changes for "initialized"